src/main/java/com/collicode/propertytracker/service/impl/TenantsServiceImpl.java

```diff
@@ -61,6 +65,18 @@ public void createTenant(TenantRequestDTO tenantRequestDTO) {
           throw StorageException.exception("ERROR WHILE CREATING TENANT");
         }
     }
+    @DeleteMapping("/{tenantId}")
+    public ResponseEntity<?> deleteTenant(@RequestHeader("X-RequestId") String requestId,
+                                             @PathVariable long tenantId) {
+        try {
+            return ResponseEntity.ok().body(TenantRepository.deleteTenant(tenantId));
```

Comment:
Controler should not interact with the repository directly ,it should be through a service.

Reply:
noted